src/wallet/rpcwallet.cpp

```diff
@@ -4837,6 +4839,359 @@ UniValue walletcreatefundedpsbt(const JSONRPCRequest& request)
+            "\nExamples:\n"
+            + HelpExampleCli("transfertoken", "\"1M72Sfpbz1BPpXFHz9m3CdqATR44Jvaydd\" 0.1")
+        );
+
+    return sendtoaddress(request);
+}
+
+/* burn token essentially means that we want to create a transaction that sends the remaining amount of colored coins to another address belonging to us. i.e send  pwallet->GetBalance()[colorId] - nValue to getnewaddress */
+static CTransactionRef BurnToken(CWallet * const pwallet, const ColorIdentifier& colorId, CAmount nValue)
+{
+    CAmount curBalance = pwallet->GetBalance()[colorId];
+
+    // Check amount
+    if (nValue <= 0)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid amount");
+
+    if (nValue > curBalance)
+        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Insufficient funds");
+
+    if (pwallet->GetBroadcastTransactions() && !g_connman) {
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
+    }
+
+    CScript scriptPubKey = CScript() << colorId.toVector() << OP_COLOR << OP_TRUE;
+    mapValue_t mapValue;
+    mapValue["comment"] = colorId.toHexString();
+
+    // Create and send the transaction
+    CReserveKey reservekey(pwallet);
+    CAmount nFeeRequired;
+    std::string strError;
+    std::vector<CRecipient> vecSend;
+    CWallet::ChangePosInOut mapChangePosRet;
+    mapChangePosRet[ColorIdentifier()] = -1;
+    CRecipient recipient = {scriptPubKey, nValue, false};
+    vecSend.push_back(recipient);
+    CTransactionRef tx;
+    CCoinControl coin_control;
+    coin_control.colorTxType = ColoredTxType::BURN;
+    //coin_control.destChange = colorDest;
+    if (!pwallet->CreateTransaction(vecSend, tx, reservekey, nFeeRequired, mapChangePosRet, strError, coin_control)) {
+        throw JSONRPCError(RPC_WALLET_ERROR, strError);
+    }
+
+    CValidationState state;
+    if (!pwallet->CommitTransaction(tx, std::move(mapValue), {} /* orderForm */, {} /*std::move(fromAccount)*/, reservekey, g_connman.get(), state)) {
+        strError = strprintf("Error: The transaction was rejected! Reason given: %s", FormatStateMessage(state));
+        throw JSONRPCError(RPC_WALLET_ERROR, strError);
+    }
+    return tx;
+}
+
+static UniValue burntoken(const JSONRPCRequest& request)
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() != 2)
+        throw std::runtime_error(
+            "burntoken \"color\" value \n"
+            "\nBurn colored coins or tokens in the wallet.\n"
+            + HelpRequiringPassphrase(pwallet) +
+            "\nArguments:\n"
+            "1. \"color\"            (string, required) The tapyrus color / token to burn.\n"
+            "2. \"value\"             (numeric or string, required) The amount to burn. eg 0.1\n"
```

Comment:
`amount` must be numeric. Token does not allow for unit conversions such as BTC<->Satoshi.